pet-store-react-hooks/src/DeletePetModal/DeleteModal.test.tsx

```diff
@@ -0,0 +1,203 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { HttpResponse, http } from 'msw';
+import { setupServer } from 'msw/node';
+
+import { DeletePetModal } from '~/DeletePetModal/DeletePetModal';
+import { apiBaseUrl } from '~/infrastructure/api-client';
+import { WaitHandle } from '~testing/WaitHandle';
+import { petKindsMap, pets, petsList } from '~testing/data';
+import { handlers } from '~testing/handlers';
+
+const server = setupServer(...handlers);
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+// Question: When i have pet for testing, is it OK to get it outside of the tests and use it in all of them?
+const petForDelete = petsList.find((x) => x.petId === 44);
+if (!petForDelete) {
+  throw new Error('Pet with petId: 44 is not found in the testing data');
+}
+
+// Question: I'm defining the callback function here and cleaning the mock data in the tests i'm using them. OK?
```

Comment:
You could, but I wouldn't. I would create the function in every test separately.

You should also call `jest.restoreAllMocks()` in a `afterEach` hook. That cleans up all mocks automatically.